### PR TITLE
Add backwards-compatibility for measures without `name` key

### PR DESCRIPTION
**Pull request by ericpgreen2**
This PR fixes the following bug:
- You have a dashboard file that includes measures without a `name` key
- You click the "Add measure" button in the Metrics UI
- The application freezes

### CODE REVIEW

File: web-common/src/features/metrics-views/metrics-internal-store.spec.ts [hunk 1]
```diff
@@ -181,4 +178,84 @@ measures:
...
+model: ""
+default_time_range: ""
+smallest_time_grain: ""
+timeseries: ""
+dimensions: []
+measures:
+`;
+    [
+      [
+        "From an old project",
```
Comment (ericpgreen2) [inline web-common/src/features/metrics-views/metrics-internal-store.spec.ts:193]: "From an old project without measure `name` keys"

File: web-common/src/features/metrics-views/metrics-internal-store.spec.ts [hunk 2]
```diff
@@ -181,4 +178,84 @@ measures:
...
+    format_preset: humanize
+    name: measure_1
+  - label: ""
+    expression: ""
+    name: measure_2
+    description: ""
+    format_preset: humanize`,
+      ],
+      [
+        "From an old project with some names",
```
Comment (ericpgreen2) [inline web-common/src/features/metrics-views/metrics-internal-store.spec.ts:221]: "From an old project with some measure `name` keys"

File: web-common/src/features/metrics-views/metrics-internal-store.ts [hunk 3]
```diff
@@ -38,6 +38,9 @@ export interface DimensionEntity {
   __ERROR__?: string;
 }
 
+const NameNumberRegex = new RegExp(/(\d+)$/);
+const MeasurePrefix = "measure";
```
Comment (ericpgreen2) [inline web-common/src/features/metrics-views/metrics-internal-store.ts:42]: "MeasureNamePrefix"

File: web-common/src/features/metrics-views/metrics-internal-store.ts [hunk 4]
```diff
@@ -38,6 +38,9 @@ export interface DimensionEntity {
   __ERROR__?: string;
 }
 
+const NameNumberRegex = new RegExp(/(\d+)$/);
```
Comment (ericpgreen2) [inline web-common/src/features/metrics-views/metrics-internal-store.ts:41]: A comment to describe where this is used and why would be helpful

**Review by AdityaHegde — APPROVED** [review]

**Pull request merged**